algorithms/src/lapack/qr_decomp.rs

```diff
@@ -0,0 +1,363 @@
+                &self,
+                mut rhs: Rhs,
+                trans: TransposeMode,
+            ) -> RlstResult<Rhs> {
+                if !check_lapack_stride(rhs.shape(), rhs.stride()) {
+                    return Err(RlstError::IncompatibleStride);
+                } else {
+                    let m = rhs.shape().0 as i32;
+                    let n = rhs.shape().1 as i32;
+                    let lda = self.data.mat.stride().1 as i32;
+                    let ldc = rhs.stride().1 as i32;
+
+                    let info = unsafe {
+                        lapacke::$lapack_qmut(
+                            lapacke::Layout::ColumnMajor,
+                            SideMode::Left as u8,
+                            trans as u8,
+                            m,
+                            n,
+                            self.data.mat.shape().1 as i32,
+                            self.data(),
+                            lda,
+                            &self.tau,
+                            rhs.data_mut(),
+                            ldc,
+                        )
+                    };
+                    if info != 0 {
+                        return Err(RlstError::LapackError(info));
+                    }
+                    Ok(rhs)
+                }
+            }
+
+            fn get_q(&self) -> RlstResult<MatrixD<Self::T>> {
+                let mut mat = rlst_mat!(Self::T, (self.shape().0, self.shape().0));
+
+                for index in 0..self.shape().0 {
+                    mat[[index, index]] = <Self::T as One>::one();
+                }
+                self.q_mult(mat, TransposeMode::NoTrans)
+            }
+
+            fn get_r(&self) -> RlstResult<MatrixD<Self::T>> {
+                let shape = self.shape();
+                let dim = std::cmp::min(shape.0, shape.1);
+                let mut mat = rlst_mat!(Self::T, (shape.0, shape.1));
+
+                for row in 0..dim {
+                    for col in row..shape.1 {
+                        mat[[row, col]] = self.data.mat[[row, col]];
+                    }
+                }
+                Ok(mat)
+            }
+
+            fn solve_qr<Rhs: RawAccessMut<T = Self::T> + Shape + Stride>(
+                &self,
+                mut rhs: Rhs,
+                trans: TransposeMode,
+            ) -> RlstResult<Rhs> {
+                rhs = self.q_mult(rhs, TransposeMode::$trans)?;
+                self.get_r()?.linalg().trisolve(
+                    rhs,
+                    TriangularType::Upper,
+                    TriangularDiagonal::NonUnit,
+                    trans,
+                )
+            }
+        }
+    };
+}
+
+qr_decomp_impl!(f32, sgeqrf, sgels, sormqr, Trans);
+qr_decomp_impl!(f64, dgeqrf, dgels, dormqr, Trans);
+qr_decomp_impl!(c32, cgeqrf, cgels, cunmqr, ConjugateTrans);
+qr_decomp_impl!(c64, zgeqrf, zgels, zunmqr, ConjugateTrans);
+
+#[cfg(test)]
+mod test {
+    #![allow(unused_imports)]
```

Comment:
Rather than allowing unused imports, can we remove the imports that are unused?

Reply:
I believe this is the only one like that - I could not figure out how to fix it. I use indexing in a macro below (lines 210-224) but don't explicitly call the Index. I had a similar one further up and ended up having to write 
`where Mat: RawAccessMut + RawAccess`
in order to get the RawAccess import "used". Do you have a recommendation on how to deal with this sort of issue?